src/js/components/Header/Header.js

```diff
@@ -1,19 +1,92 @@
-import React from 'react';
-
+import React, { useEffect, useState } from 'react';
 import { Box } from '../Box';
+import { useForwardedRef } from '../../utils';
+
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const containerRef = useForwardedRef(ref);
+  // show header as default then if scroll down showHeader is false
+  // else is scroll up showHeader is true.
+  const [height, setHeight] = useState();
+  const [width, setWidth] = useState();
+  const [right, setRight] = useState();
+  const [left, setLeft] = useState();
+  const [scrollUp, setScrollUp] = useState(false);
+
+  useEffect(() => {
+    //   // threshold inital value set to 0
+    const threshold = 0;
+    //   // number of pixels the document is currently scrolled
+    let lastScrollY = window.pageYOffset;
+    //   // run event listener callback once
+    let ticking = false;
```

Comment:
what does `ticking` mean?

Reply:
So I had the `requestAnimationFrame` to calculate the new offset once the page is rendered after the scroll is complete. 
The ticking flag was to make sure to just run the event listener callback once in each `requestAnimationFrame` 
I know it's not the best word so I changed it to `runEventListener`

open to suggestions